tests/load/gcp/loki-log-locustfile.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import json
+import random
+from datetime import datetime
+from time import time_ns
+
+from locust import constant, events, task
+from locust.contrib.fasthttp import FastHttpUser
+
+
+class LogGenerator:
+    population = [
+        "Query Execution Time:0.0022099018096924",
+        "Configuration variable date.timezone is not set, guessed timezone America/Sao_Paulo. Please set date.timezone='America/Argentina/Buenos Aires in php.ini!",
+        "Query:SELECT users.* FROM users WHERE users.id = '99bcc163-034c-ab4f-f1f3-5f7362bd45de' AND users.deleted=0 LIMIT 0,1",
+        "SugarBean constructor error: Object has not fields in dictionary. Object name was: Audit",
+        "Query:SELECT u1.first_name, u1.last_name from users u1, users u2 where u1.id = u2.reports_to_id AND u2.id = '99bcc163-034c-ab4f-f1f3-5f7362bd45de' and u1.deleted=0",
+        "Query:SELECT gcoop_salesopportunity.* FROM gcoop_salesopportunity WHERE gcoop_salesopportunity.id = '35063c55-1c51-ff9a-473f-5f7610e7ea10' AND gcoop_salesopportunity.deleted=0 LIMIT 0,1",
+        "SMTP server settings required first." "Query:SHOW INDEX FROM aow_workflow",
+        "Query:SHOW TABLES LIKE 'aow_processed'",
+        "You're using 'root' as the web-server user. This should be avoided for security reasons. Review allowed_cron_users configuration in config.php.",
+    ]
+
+    @staticmethod
+    def _stamp(line: str, num: int):
+        level = random.choice(["DEBUG", "INFO", "WARNING", "ERROR"])
+        date_time = datetime.now().isoformat()
+        return f"{date_time} - [{level}] - [{num}]: {line}"
+
+    @staticmethod
+    def _pack_log(line: str, num: int):
+        return [str(time_ns()), LogGenerator._stamp(line, num)]
+
+    @staticmethod
+    def _get_sample(num: int):
+        return [
+            LogGenerator._pack_log(random.choice(LogGenerator.population), num) for _ in range(num)
+        ]
+
+    @staticmethod
+    def generate(num: int):
+        return LogGenerator._get_sample(num)
+
+
+def timepad(pad: float):
+    """Decorator that guarantees a function won't return before some minimum time has elapsed.
+
+    Args:
+        pad: minimum time in seconds that should elapse before the wrapped function returns.
+    """
+    pad = max(0, pad)
+
+    def decorator(func):
+        import time
+        from functools import wraps
+
+        @wraps(func)
+        def inner(*args, **kwargs):
+            tic = time.time()
+            func(*args, **kwargs)
+            toc = time.time()
+            time.sleep(max(0, pad - (toc - tic)))
+
+        return inner
+
+    return decorator
+
+
+@events.init_command_line_parser.add_listener
+def _(parser):
```

Comment:
Very strange. I am not sure what you mean by throwaway. Not used ? (unlikely). Having not read the Locust docs I can only guess it is being used for some global side effect. Does that sound right ? If so any harm giving it a meaningful name that reveals the intent of the side effect ?

Reply:
Yes, it is unused. The decorator takes care of the registration so no name is needed.
Naming it may convey the wrong idea in that case.
I believe the intent is clearly communicated by the name of the decorator and the oneliner function: `parser.add_argument`; in fact I'd argue the docstring is redundant as well as a function name :)